### PR TITLE
Open Source Airflow

### PR DESCRIPTION
**Pull request by ashvjit-prophecy**

### CODE REVIEW

**Review by prongs — APPROVED** [review]

File: src/pbt/client/airflow/airflow_utility.py [hunk 1]
```diff
@@ -11,15 +12,28 @@ def create_airflow_client(fabric_id: str, project_config: ProjectConfig):
...
+            return ComposerRestClient(
+                composer.airflow_url, composer.project_id, composer.client_id, composer.key_json, composer.dag_location
+            )
 
         elif mwaa is not None:
-            return MWAARestClient(mwaa.environment_name, mwaa.region, mwaa.access_key,
-                                  mwaa.secret_key)
+            return MWAARestClient(mwaa.environment_name, mwaa.region, mwaa.access_key, mwaa.secret_key)
+
+        elif oss is not None:
```
Comment (pankajb23) [inline src/pbt/client/airflow/airflow_utility.py:26]: super nit 

`elif oss:` should work

File: src/pbt/client/airflow/mwaa.py [hunk 2]
```diff
@@ -6,29 +6,31 @@
...
 
-        self.mwaa_client = boto3.client("mwaa", region_name=region, aws_access_key_id=access_key,
-                                        aws_secret_access_key=secret_key)
+        self.mwaa_client = boto3.client(
+            "mwaa", region_name=region, aws_access_key_id=access_key, aws_secret_access_key=secret_key
+        )
 
         self.environment = self.mwaa_client.get_environment(Name=environment_name)
-        self.dag_s3_path = self.environment['Environment']['DagS3Path']
+        self.dag_s3_path = self.environment["Environment"]["DagS3Path"]
```
Comment (pankajb23) [inline src/pbt/client/airflow/mwaa.py:33]: super nit:

aren't both equivalent single quote and double quotes ?

File: src/pbt/client/airflow/open_source.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+    def upload_dag(self, dag_id: str, file_path: str):
+        try:
+            self.put_object_from_file(self.dag_home, f"{dag_id}.zip", file_path)
+        except Exception as e:
+            raise DagUploadFailedException(f"Failed to upload DAG file from path {file_path}", e)
+        # finally:
+        #     if Path(file_path).exists():
+        #         os.remove(file_path)
+
```
Comment (pankajb23) [inline src/pbt/client/airflow/open_source.py:76]: super nit:
can be removed.

File: src/pbt/deployment/pipeline.py [hunk 4]
```diff
@@ -778,31 +980,115 @@ def __init__(self, project: Project, project_config: ProjectConfig, pipeline_id:
...
+                step_id=self.pipeline_id,
+                indent=2,
+            )
+            return Either(right=True)
+
+    def exists(self) -> bool:
+        return True
+
+
+class HdfsPipelineUploader(PipelineUploader, ABC):
```
Comment (pankajb23) [inline src/pbt/deployment/pipeline.py:1028]: note to self:

time is ripe to fork new files

**Review by pankajb23 — APPROVED** [review]

File: src/pbt/client/airflow/mwaa.py (hunk 2 above)
Comment (ashvjit-prophecy) [reply src/pbt/client/airflow/mwaa.py:33]: black did it :)

File: src/pbt/client/airflow/open_source.py (hunk 3 above)
Comment (ashvjit-prophecy) [reply src/pbt/client/airflow/open_source.py:76]: In fact I'll uncomment this

**Pull request merged**